### ISSUE TITLE
Present Buildpacks to the TOC

### ISSUE THREAD

**Issue by sclevine** [cncf/toc#122]
Buildpacks replace Dockerfiles in the app development lifecycle with a higher level of abstraction. In doing so, they provide a balance of control that reduces the operational burden on developers and supports enterprise operators who manage apps at scale. Buildpacks ensure that apps meet security and compliance requirements without developer intervention. They provide automated delivery of both OS-level and application-level dependency upgrades, efficiently handling day-2 app operations that are often difficult to manage with Dockerfiles.

First conceived in 2012, buildpacks have provided an opinionated solution for building cloud apps on Heroku, Cloud Foundry, GitLab, Deis, Dokku, and other platforms. Heroku and Pivotal are actively collaborating on a next-generation buildpack API that uses existing standards (such as the OCI image format) to bring buildpacks to cloud platforms that support Docker or OCI images. This new API also incorporates feedback from years of assisting our customers operate buildpacks in production. 

We believe this project fits well within the CNCF by directly supporting the CNCF's mission statement of supporting cloud native systems. The next generation of buildpacks will aid developers and operators in packaging applications into containers (1a), allow operators to efficiently manage the infrastructure necessary to keep application dependencies updated (1b), and are available via well-defined interfaces (1c).

By becoming a CNCF project, we hope to greatly improve buildpack interoperability between platforms and attract a wide community of contributors, including buildpack creators and maintainers.

GitHub: https://github.com/buildpack/resources
License: Apache License, Version 2.0
(Website will be packs.sh.)

**Comment by duglin** [comment]
@sclevine can you elaborate on where, or what, this "project" is? The url to github you provided is just to a README. All of the links in there go to other sites. Are you proposing a project that has code - and if so, where is it? or are you suggesting we pick-up the CF code base? Or is this project a specification?  the CF spec? the heroku spec?

**Comment by caniszczyk** [comment]
RFC @cncf/toc 

If I don't hear anything against scheduling from the TOC by end of next week, how does Aug 21st work at 8am PT?

**Comment by sclevine** [comment]
@duglin We intend to contribute a specification for a new buildpack interface that adopts cloud native principles and standards, a corresponding reference implementation, and a set of tools. We don't currently plan to contribute existing buildpacks.

Apologies for the lack of clarity -- this is an early-stage project intended for the sandbox, and we plan to publish more before presenting to the TOC.

**Comment by duglin** [comment]
Great, thanks for the clarification.

**Comment by garethr** [comment]
What's a good place for folks to contribute to this effort?

**Comment by dankohn** [comment]
@sclevine Could you please speak to why Heroku and Cloud Foundry are not planning to contribute the existing buildpacks?

**Comment by caniszczyk** [comment]
@sclevine you can present to the TOC on August 21st if that works at 8am PT

**Comment by chipchilders** [comment]
@dankohn I'll let @sclevine and the Heroku folks reply as well, but generally think about buildpacks as artifacts that are designed to be varied across users and use cases. An imperfect analogy would be Dockerfiles. There's discussion in the works around a buildpack registry to help with discoverability across the ecosystem of buildpack authors.

What matters is the standardization of the API and reference implementations of that API that can support using any buildpack that conforms to the standard.

**Comment by dankohn** [comment]
I agree that you can treat buildpacks as opaque and just standardize the
interfaces for them. However, I know from experience that there are
thousands of hours of engineering invested in, say, the internals of the
Ruby or Node.js buildpacks. I would think that CF and Heroku might want to
collaborate to share that engineering rather than maintain their own
buildpacks. But perhaps that collaboration could come later.

--
Dan Kohn <dan@linuxfoundation.org>
Executive Director, Cloud Native Computing Foundation https://www.cncf.io
+1-415-233-1000 https://www.dankohn.com

**Comment by caniszczyk** [comment]
@bgrant0607 and @monadic are happy to sponsor this in the sandbox

@sclevine can you put forth a project proposal and issue a PR against the TOC repo?

**Comment by sclevine** [comment]
Thanks @bgrant0607 and @monadic! Don't hesitate to reach out to us if you have any questions about the project. We really appreciate your interest! 😄 

@caniszczyk Will do ASAP.

**Comment by gabrielyuyang** [comment]
Hi, it'd better there is a video demo. So that we can understand the whole thing piece by piece in an organized way. Now, I only vaguely know what you are proposing without any knowledge of the capabilities, usage, developments, etc.

**Comment by sclevine** [comment]
@caniszczyk proposed here: https://github.com/cncf/toc/pull/150
